app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,66 @@
+class TasksController < ApplicationController
+
+  def index
+    @tasks = Task.all
+  end
+
+  def edit
+    @task = Task.find_by(id: params[:id])
+    unless @task
+      redirect_to root_path
+    end
+  end
+
+  def new
+    @task = Task.new
+  end
+
+  def show
+    @task = Task.find( params[:id] )
```

Comment:
It's a good idea to have some sort of if statement here to handle the occasions where, you can't find the given task.   This also applies to delete, create, update, edit, etc.